pkg/authorization/api/v1beta1/conversion.go

```diff
@@ -27,6 +27,27 @@ import (
 
 func init() {
 	err := api.Scheme.AddConversionFuncs(
+		func(in *PolicyRule, out *newer.PolicyRule, s conversion.Scope) error {
+			if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
+				return err
+			}
+
+			if out.Resources == nil {
+				out.Resources = make([]string, 0)
+			}
+			if in.ResourceKinds != nil {
+				out.Resources = append(out.Resources, in.ResourceKinds...)
+			}
+
+			return nil
+		},
+		func(in *newer.PolicyRule, out *PolicyRule, s conversion.Scope) error {
```

Comment:
> if all this is doing is the default convert, can we remove it completely?

Since v1beta1 includes a field that is not present in the in-memory representation, this isn't actually doing a default conversion.  It's doing a default conversion that ignores missing fields.